node/src/main/scala/org/bitcoins/node/util/BitcoinSNodeUtil.scala

```diff
@@ -0,0 +1,51 @@
+package org.bitcoins.node.util
+
+trait BitcoinSNodeUtil {
+
+  /**
+    * Creates a unique actor name for a actor
+    * @param className
+    * @return
+    */
+  def createActorName(className: String): String = {
+    s"${className}-${System.currentTimeMillis()}"
+  }
+
+  /**
+    * Creates a unique actor name for a given class
+    * @param className
+    * @return
+    */
+  def createActorName(className: Class[_]): String =
+    createActorName(className.getSimpleName)
+}
+
+object BitcoinSNodeUtil extends BitcoinSNodeUtil {
+
+  class Mutable[A](initialValue: A) {
+    private val lock = new java.util.concurrent.locks.ReentrantReadWriteLock()
```

Comment:
you probably know better than me but isn't best practice to use `AtomicReference`?